src/components/stepper/hooks/useStepperDataReducer.ts

```
@@ -44,7 +45,12 @@ function stepperDataReducer<
 >(state: D, action: AnyAction) {
   switch (action.type) {
     case 'SET_STEP':
-      return { ...state, [action.stepId]: action.data };
+      // clone first to keep the previous state immutable
+      const mergedData = deepMerge({}, state[action.stepId], action.data);
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)